### PR TITLE
Run benchmark on Travis-CI

### PR DESCRIPTION
**Pull request by KDr2**
When your commit message contains a "[bm]", a benchmark job(**your commit** VS **master**) will run. The report will be a comment of the commit.

Please do NOT merge this PR before the travis-ci branch of https://github.com/TuringLang/TuringBenchmarks is merged.

### CODE REVIEW

File: scripts/benchmark-on-ci.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (yebai) [inline scripts/benchmark-on-ci.sh]: Perhaps consider renaming this file to `benchmarks/run-on-travis.sh`

**Comment by KDr2** [comment]
https://github.com/TuringLang/TuringBenchmarks/pull/42 is ready to merge.

After https://github.com/TuringLang/TuringBenchmarks/pull/42 is merged, I will make few trivial changes to the shell script in this PR, then we can run benchmarks on Travis:

- Your branch must contain the commits in this PR, if you have a previously created branch, you can rebase it on master to obtain the commits in this PR.
- When you push a new commit whose commt message contains the string `[bm]`, a benchemark job wll be trigerred on TravisCI to compare the performance between master and you commit.

**Comment by KDr2** [comment]
Ready to merge.

File: scripts/run-benchmarks-on-travis.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+fi
+        
+COMMIT_MSG=$(git show -s --format="%s")
+
+SANTI_BR_NAME=$(echo $CURRENT_BRANCH | sed 's/\W/_/g')
+COMMIT_SHA=$(git rev-parse  HEAD)
+TIME=$(date +%Y%m%d%H%M%S)
+BM_JOB_NAME="BMCI-${SANTI_BR_NAME}-${COMMIT_SHA:0:7}-${TIME}"
+
+if [[ $COMMIT_MSG != *"[bm]"* ]]; then
```
Comment (yebai) [inline scripts/run-benchmarks-on-travis.sh]: I think we can run benchmarking by default. The user can use “si” to skip CI together with benchmarking if needed.

File: scripts/run-benchmarks-on-travis.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (yebai) [inline scripts/run-benchmarks-on-travis.sh]: Can we turn this bash script into a Julia script? A Julia script might be easier to work with for other team members. It’s helpful for other tasks described in #822.

**Pull request merged**